src/server/game/Entities/Creature/Creature.h

```diff
@@ -102,12 +102,15 @@ class TC_GAME_API Creature : public Unit, public GridObject<Creature>, public Ma
         void InitializeReactState();
 
         // These shadow the definitions on Unit and override the default value for keepCombat
-        void SetImmuneToAll(bool apply) { Unit::SetImmuneToAll(apply, HasReactState(REACT_PASSIVE)); }
-        void SetImmuneToAll(bool apply, bool keepCombat) { Unit::SetImmuneToAll(apply, keepCombat); }
-        void SetImmuneToPC(bool apply) { Unit::SetImmuneToPC(apply, HasReactState(REACT_PASSIVE)); }
-        void SetImmuneToPC(bool apply, bool keepCombat) { Unit::SetImmuneToPC(apply, keepCombat); }
-        void SetImmuneToNPC(bool apply) { Unit::SetImmuneToNPC(apply, HasReactState(REACT_PASSIVE)); }
-        void SetImmuneToNPC(bool apply, bool keepCombat) { Unit::SetImmuneToNPC(apply, keepCombat); }
+        using Unit::IsImmuneToAll;
+        using Unit::SetImmuneToAll;
+        void SetImmuneToAll(bool apply) override { Unit::SetImmuneToAll(apply, HasReactState(REACT_PASSIVE)); }
+        using Unit::IsImmuneToPC;
+        using Unit::SetImmuneToPC;
+        void SetImmuneToPC(bool apply) override { Unit::SetImmuneToPC(apply, HasReactState(REACT_PASSIVE)); }
+        using Unit::IsImmuneToNPC;
+        using Unit::SetImmuneToNPC;
```

Comment:
Because that's how C++ scoping works. The override for the one-arg version of `SetImmuneToX` would shadow the base class definition of the two-arg version.

(The `IsImmuneToX` decl is here for clarity)